vault/dynamic_system_view.go

```diff
@@ -87,3 +88,41 @@ func (d dynamicSystemView) ReplicationState() consts.ReplicationState {
 	d.core.clusterParamsLock.RUnlock()
 	return state
 }
+
+// ResponseWrapData wraps the given data in a cubbyhole and returns the
+// token used to unwrap.
+func (d dynamicSystemView) ResponseWrapData(data map[string]interface{}, ttl time.Duration, jwt bool) (string, error) {
+	req := &logical.Request{
+		Operation: logical.CreateOperation,
+		Path:      "sys/init",
+	}
+
+	resp := &logical.Response{
+		WrapInfo: &logical.ResponseWrapInfo{
+			TTL: ttl,
+		},
+		Data: data,
+	}
+
+	if jwt {
+		resp.WrapInfo.Format = "jwt"
+	}
+
+	_, err := d.core.wrapInCubbyhole(req, resp)
+	if err != nil {
+		return "", err
+	}
+
+	return resp.WrapInfo.Token, nil
```

Comment:
I think this should return `resp.WrapInfo`. That way a caller can easily carry along the rest of the information (ttl, creation time, etc.) in their response to the user. Otherwise if it's e.g. a backend it can't actually look up those properties.